test/IRGen/extended_existential_metadata_weak.swift

```diff
@@ -0,0 +1,13 @@
+// RUN: %target-swift-frontend -emit-ir %s -target arm64-apple-macos12 | %FileCheck %s
```

Comment:
Nit: `%target-cpu-apple-macos12` would make this test run without building a bunch of modules on x86_64. Even, better `%target-swift-abi-5.5-triple` and using `@available(SwiftStdlib 5.7, *)` in the body would allow this to run on `REQUIRES: VENDOR=apple` instead of just macOS